src/groupeddataframe/splitapplycombine.jl

```diff
@@ -585,50 +670,58 @@ wrap(x::AbstractMatrix) =
     NamedTuple{Tuple(gennames(size(x, 2)))}(Tuple(view(x, :, i) for i in 1:size(x, 2)))
 wrap(x::Any) = (x1=x,)
 
+const ERROR_ROW_COUNT = "return value must not change its kind " *
+                        "(single row or variable number of rows) across groups"
+
+const ERROR_COL_COUNT = "function must return only single-column values, " *
+                        "or only multiple-column values"
+
 wrap_table(x::Any, ::Val) =
-    throw(ArgumentError("return value must not change its kind " *
-                        "(single row or variable number of rows) across groups"))
+    throw(ArgumentError(ERROR_ROW_COUNT))
 function wrap_table(x::Union{NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}},
                              AbstractDataFrame, AbstractMatrix},
                              ::Val{firstmulticol}) where firstmulticol
     if !firstmulticol
-        throw(ArgumentError("function must return only single-column values, " *
-                            "or only multiple-column values"))
+        throw(ArgumentError(ERROR_COL_COUNT))
     end
-    wrap(x)
+    return wrap(x)
 end
 
 function wrap_table(x::AbstractVector, ::Val{firstmulticol}) where firstmulticol
     if firstmulticol
-        throw(ArgumentError("function must return only single-column values, " *
-                            "or only multiple-column values"))
+        throw(ArgumentError(ERROR_COL_COUNT))
     end
-    wrap(x)
+    return wrap(x)
 end
 
 function wrap_row(x::Any, ::Val{firstmulticol}) where firstmulticol
     # NamedTuple is not possible in this branch
     if (x isa DataFrameRow) ⊻ firstmulticol
-        throw(ArgumentError("function must return only single-column values, " *
-                            "or only multiple-column values"))
+        throw(ArgumentError(ERROR_COL_COUNT))
+    end
+    return wrap(x)
+end
+
+function wrap_row(x::Union{AbstractArray{<:Any, 0}, Ref},
+                  ::Val{firstmulticol}) where firstmulticol
+    if firstmulticol
+        throw(ArgumentError(ERROR_COL_COUNT))
     end
-    wrap(x)
+    return (x1 = x[],)
```

Comment:
This is the point that it cannot (this was my initial implementation), but then `Ref([1,2,3])` gets unwrapped to a vector and in consequence we think it is a multi-row case of vectors holding `Int`s and not a single-row case where each row holds a vector (which is a correct way to handle this).

Actually this is a key functionality we provide here - it was requested on discourse - to allow for `Ref([1,2,3])` instead of `[[1,2,3]]` because the latter:
1. allocates
2. is slower
3. does not check if only one row per group is produced